src/types/inc/CodepointWidthDetector.hpp

```diff
@@ -23,13 +23,14 @@ static_assert(sizeof(unsigned int) == sizeof(wchar_t) * 2,
 class CodepointWidthDetector final
 {
 public:
-    CodepointWidthDetector() = default;
+    CodepointWidthDetector() noexcept;

```

Comment:
wasn't it implied that default ctors can't throw?

Reply:
It's not implied. It's proper practice according to the core checks to ensure that they can't and don't throw such that they can be safely copied/moved/etc.